app/controllers/contacts_controller.rb

```diff
@@ -38,9 +38,10 @@ class ContactsController < ApplicationController
   end
 
   def website_create
-    contact_recipient = params.require(:contactRecipient)
-    contact_params = params.require(contact_recipient)
-    requestor_details = current_user || params.require(:userData)
+    formValues = JSON.parse(params.require(:formValues)).deep_symbolize_keys
+    contact_recipient = formValues.fetch(:contactRecipient)
+    contact_params = formValues.fetch(contact_recipient.to_sym)
+    requestor_details = current_user || formValues.fetch(userData)
```

Comment:
After passing the JSON string through `JSON.parse`, it is a normal plain Ruby object that can be accessed as such: `formValues[:contactRecipient]` instead of `.fetch(...foo...)`

Reply:
Thing is, I want to throw error if the key is not available, hence thought of using this (just like how we use params.require(:foo) instead of params[:foo].

Reply:
Not sure if I'm happy with that.
The point why we use `params.require` is that it gives meaningful information to the client. (Since `params` is directly provided by the Rails framework, calling `.require(:foo)` when `:foo` is not set returns a proper HTTP error response code with a proper error message to the client)
With `my_hash.fetch(:some_key)` this is not the case so I don't see the big advantage of using it

Reply:
Okay, that makes sense. I've changed it. I've added a line to return bad_request in case some required argument is missed.